chain/indexer/src/streamer/mod.rs

```diff
@@ -77,14 +76,31 @@ async fn build_streamer_message(
         as near_primitives::types::NumShards;
 
     let mut shards_outcomes = fetch_outcomes(&client, block.header.hash).await?;
+    let mut state_changes = fetch_state_changes(
+        &client,
+        block.header.hash,
+        near_primitives::types::EpochId(block.header.epoch_id.clone()),
+    )
+    .await?;
     let mut indexer_shards = (0..num_shards)
-        .map(|shard_id| IndexerShard { shard_id, chunk: None, receipt_execution_outcomes: vec![] })
+        .map(|shard_id| IndexerShard {
+            shard_id,
+            chunk: None,
+            receipt_execution_outcomes: vec![],
+            state_changes: vec![],
```

Comment:
Extract the state_changes for the appropriate shard_id here immediately, since given that execution outcomes could happen when chunk was not produced, state changes will also be there outside of the chunk, so iteration over chunks won't be enough, and here is the right place to have the shard state changes populated.